tests/test_util.py

```diff
@@ -3,10 +3,32 @@
 # This file is part of gunicorn released under the MIT license.
 # See the NOTICE for more information.
 
+import pytest
+
 from gunicorn import util
 
 
 def test_parse_address():
     # Test unix socket addresses (PR #1623)
     assert util.parse_address('unix://var/run/test.sock') == 'var/run/test.sock'
     assert util.parse_address('unix:/var/run/test.sock') == '/var/run/test.sock'
+
+def test_http_date():
+    assert util.http_date(1508607753.740316) == 'Sat, 21 Oct 2017 17:42:33 GMT'
+
+
+@pytest.mark.parametrize("test_input, expected", [
+    ('1200:0000:AB00:1234:0000:2552:7777:1313', True),
+    ('1200::AB00:1234::2552:7777:1313', False),
+    ('21DA:D3:0:2F3B:2AA:FF:FE28:9C5A', True),
+    ('1200:0000:AB00:1234:O000:2552:7777:1313', False),
+])
+def test_is_ipv6(test_input, expected):
+    assert util.is_ipv6(test_input) == expected
+
+
+def test_warn(capsys):
+    expected = 'WARNING: test warn'
```

Comment:
Style nit: I'd write this as

```py
assert 'WARNING: test warn' in err
```

You may also want to test that a line starts with `'!!!'`.